grpcbp/doc.go

```diff
@@ -0,0 +1,2 @@
+// Package grpcbp provides Baseplate specific gRPC related helpers.
```

Comment:
Good point, I need to extend on the documentation here, currently not having a `doc.go` will make `staticcheck` fail.

Reply:
I disagree. It's not always obvious which file has the most exported API, always just do the package level documentation in a separated `doc.go` seems easier style to follow (and enforce if we want to enforce it).

if the package only has one non-test .go file I'm fine with writing the package doc there without a separated doc.go, but as long as it has more than one .go files I'd prefer to have a separated doc.go instead of trying to decide "which is the most important .go file to write package doc".